Sources/Quick/ExampleGroup.swift

```diff
@@ -8,6 +8,11 @@ import Foundation
 final public class ExampleGroup: NSObject {
     weak internal var parent: ExampleGroup?
     internal let hooks = ExampleHooks()
+    
+    internal var beforesStartedExecuting = false
```

Comment:
> ... I'd suggest using a smaller enum ... These particular booleans can then be computed properties on the enum, as necessary.

We can definitely do this. I agree that the state machine has some danger built into as currently implemented. Let me see what I can come up with.